File: Interview Preparation Kits/1 Month Preparation Kit/Week 1/06 - Flipping Bits/Solution_FlippingBits.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'flippingBits' function below.
#
# The function is expected to return a LONG_INTEGER.
# The function accepts LONG_INTEGER n as parameter.
#

def flippingBits(n):
    # Write your code here
    return n

def flippingBits_testgebruiktami1(n):
    max_ = -1 + 2**32
    return max_ - n

def flippingBits_matt558(n):
    return n ^ (2 ** 32 - 1)

def flippingBits_snekparti(n):
    # Write your code here
    mask = 2**32 - 1
    return n ^ mask

def flippingBits_bernardmwambua12(n):
    binary_n = "{0:b}".format(n)
    inverse_s = ''
    bin_32 = binary_n.zfill(32)
    for i in list(bin_32):
        if i == '0':
            inverse_s += '1' 
        else:
            inverse_s += '0' 
    return(int(inverse_s, 2))

if __name__ == '__main__':
    fptr = sys.stdout

    q = int(input().strip())

    for q_itr in range(q):
        n = int(input().strip())

        result = flippingBits(n)

        fptr.write(str(result) + '\n')

    fptr.close()

File: Interview Preparation Kits/1 Month Preparation Kit/Week 1/06 - Flipping Bits/Solution_FlippingBits.java
import java.io.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;

class Result_FlippingBits {

    /*
     * Complete the 'flippingBits' function below.
     *
     * The function is expected to return a LONG_INTEGER.
     * The function accepts LONG_INTEGER n as parameter.
     */

    public static long flippingBits_cfernandezcairo(long n) {
        // Write your code here
        long num = (1L<<32)-1;

        System.out.println(n^num);

        return n^num;

    }

    public static long flippingBits_linchenghao1999(long n) {
        // Write your code here
        return (long)Math.pow(2,32) + (~n);
    }

}

public class Solution_FlippingBits {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new OutputStreamWriter(System.out));

        int q = Integer.parseInt(bufferedReader.readLine().trim());

        IntStream.range(0, q).forEach(qItr -> {
            try {
                long n = Long.parseLong(bufferedReader.readLine().trim());

                long result = Result_FlippingBits.flippingBits_cfernandezcairo(n);

                bufferedWriter.write(String.valueOf(result));
                bufferedWriter.newLine();
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        });

        bufferedReader.close();
        bufferedWriter.close();
    }
}